# HG changeset patch
# User Rik <rik@octave.org>
# Date 1523855032 25200
#      Sun Apr 15 22:03:52 2018 -0700
# Node ID e5cf943ba01b5a8b0f5383e820b51e92c898a626
# Parent  7e0be1b2ccf53bc759b75896b68dbf7b421dbd2a
# Parent  32323f5e61d264b3a1efc63f6b8251fee7f189bb
maint: merge stable to default.

diff --git a/libgui/languages/pt_BR.ts b/libgui/languages/pt_BR.ts
--- a/libgui/languages/pt_BR.ts
+++ b/libgui/languages/pt_BR.ts
@@ -426,48 +426,50 @@
         <source>Octave Documentation</source>
         <translation>Documentação do Octave</translation>
     </message>
     <message>
         <location line="-326"/>
         <source>Could not copy help collection to temporary
 file. Search capabilities may be affected.
 %1</source>
-        <translation type="unfinished"></translation>
+        <translation>Não foi possível copiar a coleção de ajuda para um arquivo 
+temporário. A funcionalidade de busca pode estar afetada.
+%1</translation>
     </message>
     <message>
         <location line="+10"/>
         <source>Could not setup the data required for the
 documentation viewer. Only help texts in
 the Console Widget will be available.</source>
         <translation>Não foi possível configurar os dados necessários para o
 visualizador da documentação. Apenas os textos de ajuda
 na Janela de Comandos estarão disponíveis.</translation>
     </message>
     <message>
         <location line="+16"/>
         <source>Find:</source>
-        <translation type="unfinished"></translation>
+        <translation>Buscar:</translation>
     </message>
     <message>
         <location line="+7"/>
         <location line="+1"/>
         <source>Search forward</source>
-        <translation type="unfinished"></translation>
+        <translation>Buscar adiante</translation>
     </message>
     <message>
         <location line="+5"/>
         <location line="+1"/>
         <source>Search backward</source>
-        <translation type="unfinished"></translation>
+        <translation>Buscar para trás</translation>
     </message>
     <message>
         <location line="+25"/>
         <source>Escape</source>
-        <translation type="unfinished"></translation>
+        <translation>Esc</translation>
     </message>
     <message>
         <location line="+24"/>
         <source>Contents</source>
         <translation>Conteúdo</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -483,17 +485,17 @@ na Janela de Comandos estarão disponíveis.</translation>
     <message>
         <location line="-22"/>
         <source>Index</source>
         <translation>Índice</translation>
     </message>
     <message>
         <location line="+203"/>
         <source>Unable to register help file %1.</source>
-        <translation type="unfinished"></translation>
+        <translation>Não foi possível registrar o arquivo de ajuda %1.</translation>
     </message>
 </context>
 <context>
     <name>octave::documentation_dock_widget</name>
     <message>
         <location filename="../src/documentation-dock-widget.cc" line="+39"/>
         <source>Documentation</source>
         <translation>Documentação</translation>
@@ -761,17 +763,17 @@ para escrita: %2.</translation>
     <message>
         <location line="+5"/>
         <source>Indent Code</source>
         <translation>Indentar Código</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Convert Line Endings to &amp;Windows (CRLF)</source>
-        <translation type="unfinished"></translation>
+        <translation>Converter Finais de Linha para &amp;Windows (CRLF)</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Convert Line Endings to &amp;Unix (LF)</source>
         <translation>Converter Finais de Linha para &amp;Unix (LF)</translation>
     </message>
     <message>
         <location line="+5"/>
@@ -1108,26 +1110,33 @@ executar seu script usando o nome dele c
 Você deseja escolher outro nome?</translation>
     </message>
     <message>
         <location line="+20"/>
         <source>The current encoding %1
 can not be applied.
 
 Please select another one or cancel saving!</source>
-        <translation type="unfinished"></translation>
+        <translation>A codificação atual %1
+não pode ser aplicada.
+
+Favor selecionar outra ou cancelar o salvar!</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>The current editor contents can not be encoded
 with the selected encoding %1.
 Using it will result in data loss!
 
 Do you want to choose another encoding?</source>
-        <translation type="unfinished"></translation>
+        <translation>O conteúdo atual do editor não podem ser codificados
+com a forma de codificação selecionada %1.
+Usá-la resultará na perda de dados!
+
+Deseja escolher outra codificação?</translation>
     </message>
     <message>
         <location line="+85"/>
         <source>It seems that &apos;%1&apos; has been modified by another application. Do you want to reload it?</source>
         <translation>O arquivo &apos;%1&apos; paree ter sido modificado por outra aplicação. Deseja recarregá-lo?</translation>
     </message>
     <message>
         <location line="+29"/>
@@ -3064,22 +3073,22 @@ Você deseja usar a tecla de atalho mesmo assim, removendo-a da outra ação?</translation>
     <message>
         <location line="+4"/>
         <source>Run Menu</source>
         <translation>Menu Executar</translation>
     </message>
     <message>
         <location line="+68"/>
         <source>Import shortcuts from file ...</source>
-        <translation type="unfinished"></translation>
+        <translation>Importar atalhos de arquivo...</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Export shortcuts into file ...</source>
-        <translation type="unfinished"></translation>
+        <translation>Exportar atalhos para arquivo...</translation>
     </message>
     <message>
         <location line="-4"/>
         <location line="+5"/>
         <source>Octave Shortcut Files (*.osc);;All Files (*)</source>
         <translation>Arquivos de Teclas de Atalho do Octave (*.osc);;Todos Arquivos (*)</translation>
     </message>
     <message>
@@ -3176,37 +3185,37 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+1"/>
         <source>Edit variables.</source>
         <translation>Editar variáveis.</translation>
     </message>
     <message>
         <location line="+259"/>
         <source>Foreground</source>
-        <translation type="unfinished"></translation>
+        <translation>Primeiro plano</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Background</source>
-        <translation type="unfinished">Fundo</translation>
+        <translation>Fundo</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Selected Foreground</source>
-        <translation type="unfinished"></translation>
+        <translation>Primeiro Plano Selecionado</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Selected Background</source>
-        <translation type="unfinished"></translation>
+        <translation>Fundo Selecionado</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Alternate Background</source>
-        <translation type="unfinished"></translation>
+        <translation>Fundo Alternativo</translation>
     </message>
     <message>
         <location line="+255"/>
         <source>Variable Editor Toolbar</source>
         <translation>Barra de Ferramentas do Editor de Variáveis</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -3571,17 +3580,17 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+49"/>
         <source>Show EOL characters</source>
         <translation>Exibir caracteres de final de linha</translation>
     </message>
     <message>
         <location line="+309"/>
         <source>Wrap long lines at current window border</source>
-        <translation type="unfinished"></translation>
+        <translation>Quebrar linhas longas na borda da janela atual</translation>
     </message>
     <message>
         <location line="+475"/>
         <source>Default EOL mode</source>
         <translation>Modo padrão para final de linha</translation>
     </message>
     <message>
         <location line="-420"/>
@@ -3963,17 +3972,17 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+13"/>
         <source>Restore editor tabs from previous session on startup or when editor is shown again after closing</source>
         <translation>Restaurar abas no editor da sessão anterior durante a inicialização ou quando o editor é exibido novamente após ser fechado</translation>
     </message>
     <message>
         <location line="+80"/>
         <source>Text encoding used for loading and saving</source>
-        <translation>Codificação de texto utilizada para abrir e salvar</translation>
+        <translation>Codificação de texto utilizada para carregar e salvar</translation>
     </message>
     <message>
         <location line="+25"/>
         <source>Close all files when the editor widget is closed/hidden</source>
         <translation>Fechar todos os arquivos quando a janela do editor é fechada/ocultada</translation>
     </message>
     <message>
         <location line="+55"/>
diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -1334,16 +1334,18 @@ namespace octave
                 glVertex3d (xPlaneN, yPlaneN, zPlaneN);
                 glVertex3d (xPlaneN, yPlaneN, zPlane);
               }
           }
       }
 
     glEnd ();
 
+    set_linestyle ("-");  // Disable LineStipple
+
 #else
 
     octave_unused_parameter (props);
 
     // This shouldn't happen because construction of opengl_renderer
     // objects is supposed to be impossible if OpenGL is not available.
 
     panic_impossible ();
@@ -2126,18 +2128,16 @@ namespace octave
 
     if (! is2D || props.layer_is ("bottom"))
       {
         draw_axes_grids (props);
         if (props.get_tag () != "legend" || props.get_box () != "off")
           draw_axes_boxes (props);
       }
 
-    set_linestyle ("-");  // Disable LineStipple
-
     set_clipbox (x_min, x_max, y_min, y_max, z_min, z_max);
 
     draw_axes_children (props);
 
     if (is2D && props.layer_is ("top"))
       {
         draw_axes_grids (props);
         if (props.get_tag () != "legend" || props.get_box () != "off")
@@ -3578,16 +3578,18 @@ namespace octave
 
         glBegin (GL_LINE_STRIP);
         glVertex2d (x0, y0);
         glVertex2d (x1, y0);
         glVertex2d (x1, y1);
         glVertex2d (x0, y1);
         glVertex2d (x0, y0);
         glEnd ();
+
+        set_linestyle ("-");
       }
 
     // Restore previous coordinate system
     glPopMatrix();
     glMatrixMode (GL_PROJECTION);
     glPopMatrix();
 
 #else
